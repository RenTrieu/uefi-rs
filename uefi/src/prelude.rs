// SPDX-License-Identifier: MIT OR Apache-2.0

//! This module is used to simplify importing the most common UEFI types.
//!
//! This includes the system table modules, `Status` codes, etc.

pub use crate::{
    boot, cstr16, cstr8, entry, runtime, system, Handle, ResultExt, Status, StatusExt,
};
