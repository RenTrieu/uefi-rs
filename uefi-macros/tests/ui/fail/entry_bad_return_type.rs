#![no_main]

use uefi::prelude::*;

#[entry]
fn main() -> bool {
    false
}
