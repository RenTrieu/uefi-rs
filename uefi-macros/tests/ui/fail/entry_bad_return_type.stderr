error[E0308]: mismatched types
 --> tests/ui/fail/entry_bad_return_type.rs:6:1
  |
6 | fn main() -> bool {
  | ^^^^^^^^^^^^^^^^^ expected `Status`, found `bool`
  |
  = note: expected fn pointer `extern "efiapi" fn(Handle, *const c_void) -> Status`
             found fn pointer `extern "efiapi" fn(Handle, *const c_void) -> bool`
